### DISCUSSION TITLE
Setting timestamps to true returns error and isoDates does nothing to date fields

### DISCUSSION THREAD

**Discussion by jonathanalberghini** [sensedeep/dynamodb-onetable#166] [Q&A] [answered]
Error
I get this error on Unsupported type passed: Wed Oct 27 2021 12:02:24 GMT-0400 (Eastern Daylight Time). Pass options.convertClassInstanceToMap=true to marshall typeof object as map attribute.

The code is below
```
    "@aws-sdk/client-dynamodb": "3.38.0",
    "@aws-sdk/util-dynamodb": "3.38.0",
    "dynamodb-onetable": "1.9.4",

const { Dynamo } = require('dynamodb-onetable/Dynamo')
// eslint-disable-next-line no-unused-vars
const { Model, Table } = require('dynamodb-onetable')
const { DynamoDBClient } = require('@aws-sdk/client-dynamodb')

const { schema } = require('../dbSchema/schema')

const insertJob = async (ts) => {
  try {
    const client = new Dynamo({ client: new DynamoDBClient({ region: 'us-east-1' }) })

    // eslint-disable-next-line no-unused-vars
    const table = new Table({
      client: client,
      name: process.env.DYNAMODB_TABLE,
      schema: schema,
      delimiter: '#',
      // crypto: Crypto,
      uuid: 'ulid',
      // logger: true,
     timestamps: true,
      isoDates: true
    })
    const Job = table.getModel('job')
    const job = await Job.create({ name: 'Aclme jkljlRlockets', lambda: 'somefunc' })
    return job
  } catch (err) {
    console.log(err.message)
  }
}
module.exports = {
  insertJob
}

/*
    Overview schema
*/
const Match = {
  email: /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
  name: /^[a-z0-9 ,.'-]+$/i,
  address: /[a-z0-9 ,.-]+$/,
  zip: /^\d{5}(?:[-\s]\d{4})?$/,
  phone: /^[+]*[(]{0,1}[0-9]{1,4}[)]{0,1}[-\s\./0-9]*$/,
  ulid: /^[0-9A-Z]{26}$/i,
  uuid: /^[0-9A-F]{32}$/i
}

const schema = {
  version: '1.0.0',
  indexes: {
    primary: { hash: 'pk', sort: 'sk' },
    gs1: { hash: 'gs1pk', sort: 'gs1sk', project: ['gs1pk', 'gs1sk'], follow: true }
  },
  models: {
    job: {
      // eslint-disable-next-line no-template-curly-in-string
      pk: { type: String, value: '${_type}#${id}' },
      // eslint-disable-next-line no-template-curly-in-string
      sk: { type: String, value: '${_type}#' },
      id: { type: String, uuid: true, validate: Match.ulid },
      name: { type: String, required: true, unique: true, validate: Match.name },
      status: { type: String, required: true, default: 'active', enum: ['active', 'inactive'] },
      lambda: { type: String, required: true },
      validateData: { type: String, required: false, default: '{}' },
      runData: { type: String, required: false, default: '{}' },

      // eslint-disable-next-line no-template-curly-in-string
      gs1pk: { type: String, value: '${_type}#' },
      // eslint-disable-next-line no-template-curly-in-string
      gs1sk: { type: String, value: '${_type}#${name}#${id}' }
    }
  }
}

module.exports = {
  schema
}
```

**Comment by jonathanalberghini** [comment]
"SyntaxError: Unexpected token ')'\n    at insertJob (/Users/jalberghini/Documents/src/tag-processor-api/logic/jobs.js:28:5)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n    at async module.exports.createJobAdmin (/Users/jalberghini/Documents/src/tag-processor-api/lambdas/tagJobAdmin.js:11:17)\n    at async InProcessRunner.run (/Users/jalberghini/Documents/src/tag-processor-api/node_modules/serverless-offline/dist/lambda/handler-runner/in-process-runner/InProcessRunner.js:211:24)\n    at async LambdaFunction.runHandler (/Users/jalberghini/Documents/src/tag-processor-api/node_modules/serverless-offline/dist/lambda/LambdaFunction.js:355:20)\n    at async hapiHandler (/Users/jalberghini/Documents/src/tag-processor-api/node_modules/serverless-offline/dist/events/http/HttpServer.js:601:18)\n    at async module.exports.internals.Manager.execute (/Users/jalberghini/Documents/src/tag-processor-api/node_modules/@hapi/hapi/lib/toolkit.js:45:28)\n    at async Object.internals.handler (/Users/jalberghini/Documents/src/tag-processor-api/node_modules/@hapi/hapi/lib/handler.js:46:20)\n    at async exports.execute (/Users/jalberghini/Documents/src/tag-processor-api/node_modules/@hapi/hapi/lib/handler.js:31:20)\n    at async Request._lifecycle (/Users/jalberghini/Documents/src/tag-processor-api/node_modules/@hapi/hapi/lib/request.js:312:32)\n    at async Request._execute (/Users/jalberghini/Documents/src/tag-processor-api/node_modules/@hapi/hapi/lib/request.js:221:9)"

**Comment by mobsense** [comment]
Can you please follow the issue template so we can better assist you.

**Comment by jonathanalberghini** [comment]
I made the debug.ts and used the same code  and everything works. So there must be something in my project or a conflicting package that is throwing it off. Sorry.

**Reply by mobsense** [reply]
No worries. I'd be interested to know if you ever find out what the difference was.

**Comment by jonathanalberghini** [comment]
I am just not going to use timestamps.
But in my code in the cjs dist directory 
in Model.js line 1029 the created field value is undefined.
```
            if (value !== undefined) {
                rec[name] = this.transformWriteAttribute(op, field, value, params);
            }
```

The field get set when I debug through after in PutItem function line 740.
This only happens when there is a unique field in the schema.

**Comment by jonathanalberghini** [comment]
Thanks

**Comment by jonathanalberghini** [comment]
Here is a video of the debug session.

https://user-images.githubusercontent.com/24964313/139288676-d3bb9a69-3ede-4024-a491-177e7c68ec0c.mp4

**Comment by jonathanalberghini** [comment]
Most likely this is a timing issue with async handler

**Comment by mobsense** [comment]
Your code is using 1.9.4. Can you retest with the latest release?

You posted:
```
    "@aws-sdk/client-dynamodb": "3.38.0",
    "@aws-sdk/util-dynamodb": "3.38.0",
    "dynamodb-onetable": "1.9.4",
```

**Comment by jonathanalberghini** [comment, marked as answer]
I upgraded to 
    "@aws-sdk/client-dynamodb": "3.38.0",
    "@aws-sdk/util-dynamodb": "3.38.0",
    "dynamodb-onetable": "2.0.2",
    
And it works perfect!!!!
I am so sorry to waste your time.
I had gotten the version from the samples.
I thought for a second I was going crazy!! :)

**Reply by mobsense** [reply]
Great to hear
